### ISSUE TITLE
Using WASI from wasmer cli

### ISSUE THREAD

**Issue by ljulliar** [wasmerio/wasmer#2978]
Hi all,

I'm generating WASM module from a compiler of my own. In one of the module I wrote I'd like to use the WASI interface. I am therefore importing the various WASI functions from the module wasi_unstable but when I try to execute this model with the wasmer runtime through the command line I get the following error:

```
error: failed to run `../rlang_apps/base64_benchmark/rlang_base64_benchmark.wasm`
╰─▶ 1: Error while importing "wasi_unstable"."args_sizes_get": unknown import. Expected Function(FunctionType { params: [I32, I32], results: [I32] })
```
Is it that I am importing the wrong module name ? Or is there an option on the wasmer CLI to "activate" the WASI module at execution time ?

Thanks for your help.

**Comment by jcaesar** [comment]
In general, this should work, WASI is active by default. One suspicion though: You're not by any chance using `wasmer run --invoke _foobar`? Because using `--invoke` deactivates WASI.

For reference: You can save this file as `test.wat` and execute it with `wasmer run test.wat` just fine (not that it does anything):
```wat
(module
  (import "wasi_unstable" "args_sizes_get"
    (func $__wasi_args_sizes_get (param i32 i32) (result i32)))
  (func $_start)
  (memory 1)
  (export "memory" (memory 0))
  (export "_start" (func $_start))
)
```

**Comment by ljulliar** [comment]
Ah, Ah... Nice catch. yes I'm using the --invoke. Let me try without that.

**Comment by jcaesar** [comment]
Possibly related to #1716.

**Comment by jcaesar** [comment]
@fschutt Thank you for fixing this. For reference, an easier way to reproduce this problem from a rust-compiled binary is
```rust
#[no_mangle]
pub extern "system" fn runme() {
    println!("Hello, world!");
}
```
You'll have to compile that as a `[lib]` with `crate-type = ["cdylib"]`.

**Comment by syrusakbary** [comment]
Thanks for opening the issue @ljulliar. It seems this may be fixed by #2997.
If not, it might be caused by not attaching the WASI imports when instantiating the instance.

If you have code we can look at that will help on debugging the issue!

**Comment by fschutt** [comment]
@syrusakbary it is fixed by #2997. See my unit test: https://github.com/wasmerio/wasmer/pull/2997/files#diff-656fd19689c83035f9a1f76eb557544908fcd6399a3784cd924c4e69e3d244afR68
